### PR TITLE
[bunsen] Detect server crash and show an error message

### PR DESCRIPTION
**Pull request by grin**
The window prompt logic is partially based on the logic reverted in:
https://github.com/twosigma/beaker-notebook/commit/fc9ac9d307ef64151a45652f0d03f3fb226da4ef

### CODE REVIEW

File: core/src/main/web/app/mainapp/mainapp.js [hunk 1]
```diff
@@ -616,8 +641,18 @@
               disconnected = true;
             },
             onReconnected: function() {
-              bkSessionManager.reconnectEvaluators();
-              disconnected = false;
+              bkSessionManager.isSessionValid()
```
Comment (mforsyth) [inline core/src/main/web/app/mainapp/mainapp.js:644]: So, what well isSessionValid() resolve to if the call to ../beaker/rest/session-backup/getExistingSessions returns a 503?

Comment (grin) [inline core/src/main/web/app/mainapp/mainapp.js:644]: Why would it return 503? This code is evaluated after the client was successfully reconnected to the server...specifically, to the new server instance that was started by marathon after the original instance crashed.

Comment (mforsyth) [inline core/src/main/web/app/mainapp/mainapp.js:644]: Oh, I see.  So when a beaker instance crashes, the client is already successfully reconnecting to a new instance?  (it's just that session is not valid anymore)?

Comment (grin) [inline core/src/main/web/app/mainapp/mainapp.js:644]: Yes, the session becomes invalid. To be more precise: there is a short period in between the server crash and the launch of a new instance where the client is disconnected and during that time it polls for a connection. When the connection is finally made the client validates the session.

Comment (mforsyth) [inline core/src/main/web/app/mainapp/mainapp.js:644]: Ah, I see.  Thanks for explaining :+1:

**Comment by buob** [comment]
:+1:

**Pull request merged**